### PR TITLE
Automerge backported PRs

### PR DESCRIPTION
**Pull request by alxnddr**
## Changes

Update the backport workflows to automerge autobackported PRs.

### Required changes:
- Enable automerge on the repo level
- Make `release-*` branches protected
    - Mark checks as required that are considered by us as required

### How to verify
- Try the workflow on your repo or check [my fork](https://github.com/alxnddr/metabase/pull/125)

### CODE REVIEW

**Review by diogormendes — APPROVED** [review]

**Comment by camsaul** [comment]
I don't know if automerge makes sense generally.

A lot of times I will review a PR and leave a few style notes or point out a typo in a docstring/comment but give it a :heavy_check_mark: anyway so people don't have to wait another day for another review cycle after making a tiny change.

If we could do automerge only for backports tho I think it would make sense.

**Comment by alxnddr** [comment]
> I don't know if automerge makes sense generally.
> 
> A lot of times I will review a PR and leave a few style notes or point out a typo in a docstring/comment but give it a ✔️ anyway so people don't have to wait another day for another review cycle after making a tiny change.
> 
> If we could do automerge only for backports tho I think it would make sense.

The automerge feature does not automatically merge PRs once they have approvals and green CI checks. It is a way for a developer to avoid waiting and manually checking of their PRs when they actually do want to merge it. So in your case the engineer after your approval will address the tiny changes you are suggesting and instead of waiting when `Squash and Merge` button becomes enabled, the engineer will press "Squash and Merge via automerge" to leave merging to GH once checks are green.

Please have a look at [the way it works](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request).

**Pull request merged**
